### ISSUE TITLE
Knots by Wikipedia

### ISSUE THREAD

**Issue by Popolechien** [openzim/zim-requests#344]
- Website URL: all articles from [WikiProject Knots](https://en.wikipedia.org/wiki/Wikipedia:WikiProject_Knots)
- License: **CC-by-sa ** 
- Desired ZIM Title: Knots by Wikipedia
- Desired ZIM Description: A subset of the encyclopedia dedicated to knots
- Desired ZIM Icon –png (URL or attach one): https://upload.wikimedia.org/wikipedia/commons/3/39/Flemishbend.jpg
- Language (ISO 639-3): **eng**
- Desired Main Page (homepage, if different from website URL): Use the conventional tile display
- Is this a MediaWiki?: yes

**Comment by benoit74** [comment]
Should we rely on wp1_selection_tools? But I don't see this project in https://download.openzim.org/wp1/enwiki/projects/ ...

Should we rely on wp1.openzim.org? But again this project is not in the list of projects ...

@Popolechien it looks like this is not really a wiki project ... or I don't get what ends up as a wiki project in WP1.

Category https://en.wikipedia.org/wiki/Category:Knots seems promising, we could use https://petscan.wmcloud.org/?psid=35592065 and wp1.openzim.org to generate proper list to use in mwoflliner.

WDYT?

**Comment by Popolechien** [comment]
> Category https://en.wikipedia.org/wiki/Category:Knots seems promising, we could use https://petscan.wmcloud.org/?psid=35592065 and wp1.openzim.org to generate proper list to use in mwoflliner.

LGTM

**Comment by benoit74** [comment]
How do we tackle such "WP1-based" selections:
- under which user do we create the selection so that anyone can update it when needed?
- how do we ensure the selection is regularly updated? (I don't see an auto-update feature in WP1 - yet?)

Can you confirm we have no answers to both of these questions and I should implement some process around it?

**Comment by Popolechien** [comment]
I can confirm we have no answer for either of these questions.
This being said, I now recall that the ask for this specific selection was because this content is very much Prepper-related. 

Maybe @anupulu has something to say about whether this is a good idea or not (or if it should be tested somehow before going to PROD).

**Comment by kelson42** [comment]
> * under which user do we create the selection so that anyone can update it when needed?

This is an open question which has to be sorted quickly. Recommended approaches:
* Create a generic "kiwix" or "kiwixbot" user at Wikimedia level (to have an account for WP1)
* Wait that we have OIDC configured on ory and integrated Ory IAM to WP1 (my preference)

> * how do we ensure the selection is regularly updated? (I don't see an auto-update feature in WP1 - yet?)

For the moment there is no solution to auto-update either selections or ZIM files on wp1.openzim.org. The idea behind it was that this is not really something we want to have - at least for the moment - because wp1.openzim.org is end-user oriented.

That said, we are currently integrated wp1_selection_tools to WP1 (in Python) and these selections has to be updated once a month. The problem with that is that WikiProject:Knots is not declared to the WP1 bot and therefore not recognized as such there. My recommendation is to do that now. Should be not more work than one hour.

The other solution would to introduce periodic/update of user based selections... which is a more complex topic.

**Comment by benoit74** [comment]
> The problem with that is that WikiProject:Knots is not declared to the WP1 bot and therefore not recognized as such there. My recommendation is to do that now. Should be not more work than one hour.

Do we have any documentation on how to do this? Or who knows how to do this? This would definitely be my preference as well, especially since it means we do not have to answer the other point for now + it is more "standard".

**Comment by kelson42** [comment]
> > The problem with that is that WikiProject:Knots is not declared to the WP1 bot and therefore not recognized as such there. My recommendation is to do that now. Should be not more work than one hour.
> 
> Do we have any documentation on how to do this? Or who knows how to do this? This would definitely be my preference as well, especially since it means we do not have to answer the other point for now + it is more "standard".

@audiodude @walkerma I guess this is a question for which you might have an answer?

**Comment by audiodude** [comment]
It is a WikiProject

However there is no [Knots articles by quality](https://en.wikipedia.org/w/index.php?title=Category:Knots_articles_by_quality) category, which is why the project doesn't appear in WP1.

The instructions for setting up the bot are here: https://en.wikipedia.org/wiki/Wikipedia:Version_1.0_Editorial_Team/Using_the_bot#Setting_up_for_the_bot

**Comment by benoit74** [comment]
@kelson42 looking at the documentation, this looks like a lot of work for something which could be solved by a pet scan query. Who will assign the various category to all pages, especially class-oriented categories (I honestly have few time and not enough experience at all for that)? And more importantly, who will maintain this in the long run?

**Comment by audiodude** [comment]
@benoit74 

It is a lot of work. That's why WikiProjects are advised to be tackled by a dedicated team of experienced editors. Pages are automatically added to the proper categories when they are assesed and the template is added to the talk page. So you don't need to edit category pages directly.

For example, look at the talk page for [Holy Spirit](https://en.wikipedia.org/wiki/Talk:Holy_Spirit). It has the following templates, which cause the talk page to be included in the proper categories. When the WP 1.0 bot runs, it resolves the talk pages to the actual article namespace/pages.

```
{{WikiProject Catholicism|importance=Top}}
{{WikiProject Islam |importance=mid}}
{{WikiProject Bahá'í Faith |importance=mid}}
{{WikiProject Religion|importance=Top|Interfaith=yes}}
{{WikiProject Mythology|importance=Top}}
```

> Who will assign the various category to all pages, especially class-oriented categories (I honestly have few time and not enough experience at all for that)? And more importantly, who will maintain this in the long run?

Like I said, this should be done by a dedicated team of volunteer editors who are committed to the project.

**Comment by benoit74** [comment]
@audiodude thank for confirming!

So this moves us back to initial idea which I will implement:
- create a selection on WP1 with a central account (looks like we already have a [KiwixOffline](https://meta.wikimedia.org/wiki/Special:CentralAuth/KiwixOffline) user) so that anyone can "access" this selection settings
- use this selection on the recipe, set for quarterly update
- manually trigger a selection refresh every quarter until we have something automated possible

**Comment by benoit74** [comment]
Or it is possible to create an "always updating" link to petscan result in raw (TSV) format?

**Comment by benoit74** [comment]
Yes it is possible: https://petscan.wmcloud.org/?format=plain&psid=35592065 ; I will simply use this in zimfarm recipe.

**Comment by benoit74** [comment]
Recipe: https://farm.openzim.org/recipes/wikipedia_en_knots

**Comment by benoit74** [comment]
@Popolechien can you please review ZIMs of https://farm.openzim.org/pipeline/dab64f9d-0f83-46fc-b5d0-48215750e0bd, available at https://dev.library.kiwix.org/#lang=eng&q=knots ~~(ignore the ZIMs with wrong names here, they should be gone soon)~~.

**Comment by benoit74** [comment]
Open question: does it really make sense to build a nopic for a selection about knots?

**Comment by Popolechien** [comment]
Looks good, thank you. There would be little value in a nopic selection seeing how many entries have literal pictures explaining how to build these knots (file size is also pretty small).

**Comment by benoit74** [comment]
File is ready in prod: https://library.kiwix.org/#lang=&category=wikipedia&q=knot
Recipe is scheduled for monthly update.
